app/controllers/main/routes.py

```diff
@@ -45,42 +46,76 @@ def events(selectedTerm=None):
         currentTime = currentTime,
         user = g.current_user)
 
-@main_bp.route('/profile/<username>', methods = ['GET'])
-def profilePage(username):
-    if not g.current_user.isCeltsAdmin and not g.current_user.isCeltsStudentStaff and g.current_user.username != username:
-        return "Access Denied", 403
+@main_bp.route('/profile/<username>', methods=['GET'])
+def viewVolunteersProfile(username):
+    """
+    This function displays the information of a volunteer to the user
+    """
+    try:
+        User.get(User.username == username)
+    except Exception as e:
+        print(e)
+        return "User does not exist", 404
+    if (g.current_user.username == username) or g.current_user.isAdmin:
+         upcomingEvents = getUpcomingEventsForUser(username)
+         programs = Program.select()
+         interests = Interest.select().where(Interest.user == username)
+         programsInterested = [interest.program for interest in interests]
+         # trainingChecklist = {}
+         # for program in programs:
+         #     trainingChecklist[program.id] = trainedParticipants(program.id)
+         eligibilityTable = []
+         for program in programs:
+              notes = ProgramBan.select().where(ProgramBan.user == username,
+                                                ProgramBan.program == program,
+                                                ProgramBan.endDate > datetime.datetime.now())
+              noteForDict = "None"
+              for j in notes:
+                  noteForDict = j.banNote.noteContent
```

Comment:
This is a long way around to get only the most recent note, and it relies on the query being sorted implicitly. I'd just get the most recent straight out of the query, or even use a slice of the list of all notes (like `list(notes)[-1].banNote.noteContent`)